--- a/library/sysinfo.c
+++ b/library/sysinfo.c
@@ -93,21 +93,16 @@ PROCPS_EXPORT int procps_loadavg(
         double *restrict av15)
 {
     double avg_1=0, avg_5=0, avg_15=0;
-    locale_t tmplocale;
     int retval=0;
     FILE *fp;
 
     if ((fp = fopen(LOADAVG_FILE, "r")) == NULL)
         return -errno;
 
-    tmplocale = newlocale(LC_NUMERIC_MASK, "C", (locale_t)0);
-    uselocale(tmplocale);
     if (fscanf(fp, "%lf %lf %lf", &avg_1, &avg_5, &avg_15) < 3)
         retval = -ERANGE;
 
     fclose(fp);
-    uselocale(LC_GLOBAL_LOCALE);
-    freelocale(tmplocale);
     SET_IF_DESIRED(av1,  avg_1);
     SET_IF_DESIRED(av5,  avg_5);
     SET_IF_DESIRED(av15, avg_15);
