--- a/library/sysinfo.c
+++ b/library/sysinfo.c
@@ -94,7 +94,9 @@ PROCPS_EXPORT int procps_loadavg(
     FILE *fp;
     char *savelocale;
 
+#if defined(__GLIBC__)
     locale_t tmplocale;
+#endif
 
     if (!(savelocale = setlocale(LC_NUMERIC, NULL)))
         return -errno;
@@ -102,6 +104,8 @@ PROCPS_EXPORT int procps_loadavg(
         return -errno;
 
+#if defined(__GLIBC__)
     tmplocale = newlocale(LC_NUMERIC_MASK, "C", (locale_t)0);
     uselocale(tmplocale);
+#endif
 
     if (!(fp = fopen(LOADAVG_FILE, "r")))
@@ -110,6 +114,8 @@ PROCPS_EXPORT int procps_loadavg(
     fscanf(fp, "%f %f %f", &av1, &av5, &av15);
     fclose(fp);
 
+#if defined(__GLIBC__)
     uselocale(LC_GLOBAL_LOCALE);
     freelocale(tmplocale);
+#endif
     setlocale(LC_NUMERIC, savelocale);
 
     return 0;
