--- a/library/sysinfo.c
+++ b/library/sysinfo.c
@@ -93,6 +93,7 @@ PROCPS_EXPORT int procps_loadavg(
     FILE *fp;
     char *savelocale;
 
+#if defined(__GLIBC__)
     locale_t tmplocale;
 
     if (!(savelocale = setlocale(LC_NUMERIC, NULL)))
@@ -102,10 +103,17 @@ PROCPS_EXPORT int procps_loadavg(
 
     tmplocale = newlocale(LC_NUMERIC_MASK, "C", (locale_t)0);
     uselocale(tmplocale);
+#else
+    if (!(savelocale = setlocale(LC_NUMERIC, NULL)))
+        return -errno;
+    if (!setlocale(LC_NUMERIC, "C"))
+        return -errno;
+#endif
 
     if (!(fp = fopen(LOADAVG_FILE, "r")))
         return -errno;
     fscanf(fp, "%f %f %f", &av1, &av5, &av15);
     fclose(fp);
 
+#if defined(__GLIBC__)
     uselocale(LC_GLOBAL_LOCALE);
     freelocale(tmplocale);
+#endif
     setlocale(LC_NUMERIC, savelocale);
 
     return 0;
